First fix for https://github.com/OWASP/owasp-masvs/issues/302 (#1465)
OWASP/owasp-mstg: Document/0x04i-Testing-user-interaction.md

A lot has happened lately in terms of responsibilities that developers have to educate users on what they need to know.
This has shifted especially with the introduction of the [General Data Protection Regulation (GDPR)](https://gdpr-info.eu/ "GDPR") in Europe. Ever since then, it is best to educate users on what is happening with their private data and why.
Additionally, it is a good practice to inform the user about how he can best use the application to ensure a secure processing of his information.

> Next, a user should be informed on what type of device-data the app will access, whether that is PII or not.

Reviewer: ```suggestion
Next, a user should be informed on what type of device data the app will access, whether that is PII or not.
```

- **Specific credentials**: When a user gets a recovery code, a password or a pin from the application (or sets one), instruct the user to never share this with anyone else and that only the app will request it.
- **Application distribution**: In case of a high-risk application it is recommended to communicate what the official way of distributing the app is. Otherwise, users might use other channels in which they download a compromised version of the application.

> #### Access to device-data

Reviewer: Style guide: Chicago style
```suggestion
#### Access to Device Data
```

> Allthough partially covered by the Google Play Store and the Apple App Store, you still need to explain to the user which services your app consumes and why. For instance: does your app require access to the contact list? Explain the user why. Does your app need access to location services of the device? Explain the user why. Does your app use device-identifiers to identify the device? Explain the user why your app needs to do that.

Reviewer 2: Maybe add something like: 

Good examples for proper explanations can be found in the: 
- [Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/ "Apple Human Interface Guidelines") and
- in the [Android App permissions best practices](https://developer.android.com/training/permissions/requesting.html#explain "Android App permissions best practices").

> Allthough partially covered by the Google Play Store and the Apple App Store, you still need to explain to the user which services your app consumes and why. For instance: does your app require access to the contact list? Explain the user why. Does your app need access to location services of the device? Explain the user why. Does your app use device-identifiers to identify the device? Explain the user why your app needs to do that.

Reviewer: ```suggestion
Although partially covered by the Google Play Store and the Apple App Store, you still need to explain to the user which services your app consumes and why. For instance:

- Does your app require access to the contact list?
- Does your app need access to location services of the device?
- Does your app use device identifiers to identify the device?

Explain the user why your app needs to do this kind of things.
```

Reviewer: made it a bit more fluid with the list and removed duplicated sentences.

> #### Other information you have to share (OSS information)

Reviewer: Style guide: Chicago style
```suggestion
#### Other Information You Have to Share (OSS Information)
```

> #### Website to help with understanding licenses

Reviewer: ```suggestion
#### Website to Help with Understanding Licenses
```

- <https://tldrlegal.com/>

> #### Guidance on permission requesting

Reviewer: ```suggestion
#### Guidance on Permission Requesting
```
